### PR TITLE
Feature/#3 better display of language icons

### LINKED ISSUE #3: Better display of language icons
Refined tasks:
- [x] Episode List (Box view)
- [x] Episode List (List view)
- [x] Episode List (Compact view)
- [ ] Player
---

Something else: Dub and sub icons are displayed close together without any relation to each other in the episode list. In the selection below an episode it is a bit random. in that example we have (scheme: *dub (sub)*):
```
jap (en)
en
jap (de)
de
```
image

I see what you did there: ordering by "seconday" (not jap) language  with dub > sub.
Another idea for an rearrangement would be to put subs for the same dub below each other.
In episode list and episode lang selection, both would profit from that more matrix style display mode. the correlation from dub and sub is shown better and it offers more possibilites to combine dubs and subs (which may be interesting in the future if more languages will come).
image

(Those images are paint edits so please don´t blame me xd)

### PR DESCRIPTION
**Pull request by Serraniel**
Screenshots for the improvments are added into #3 .

### CODE REVIEW

**Comment by Serraniel** [comment]
I just noticed that the icons below the player also are part of the issue. So this should not be merged yet. However, I think the algorithm itself hopefully will not be affected a lot so I think the current state already can be reviewed and the merge being delayed until the issue is completed.

- [x] add implementation for the player.

**Comment by Serraniel** [comment]
Nvm, I created #58 as successor for the realignment of the language selection below the player as it is a differnt at all and would blow up the pull request with additional code for a different feature.

File: src/javascript/enhancements/languageDisplay.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+function updateLanguageDisplayListMode(node) {
+    // last column with flags
+    let col = node.querySelector('h3.layout-align-end-center');
+
+    if (typeof col === 'undefined' || col.awpManipulated) {
+        return;
+    }
+
+    dopUpdateLanguageDisplay(col, false);
```
Comment (kaffem) [inline src/javascript/enhancements/languageDisplay.js]:
```suggestion
    doUpdateLanguageDisplay(col, false);
```

File: src/javascript/enhancements/languageDisplay.js [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+function updateLanguageDisplayBoxMode(node) {
+    // last column with flags
+    let col = node.querySelector('div.layout-align-end-start');
+
+    if (typeof col === 'undefined' || col.awpManipulated) {
+        return;
+    }
+
+    dopUpdateLanguageDisplay(col, true);
```
Comment (kaffem) [inline src/javascript/enhancements/languageDisplay.js]:
```suggestion
    doUpdateLanguageDisplay(col, true);
```

File: src/javascript/enhancements/languageDisplay.js [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+}
+
+
+function dopUpdateLanguageDisplay(parent, isBoxedModed) {
```
Comment (kaffem) [inline src/javascript/enhancements/languageDisplay.js]:
```suggestion
function doUpdateLanguageDisplay(parent, isBoxedModed) {
```

**Review by kaffem — APPROVED** [review]

**Comment by Serraniel** [comment]
Applied all of the typo fixes 👌

**Review by kaffem — APPROVED** [review]

**Pull request merged**